File: sources/win-ui-frontend/Fin-Manager-v2/DTO/UpdateFinanceAccountDto.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace Fin_Manager_v2.DTO
{
    public class UpdateFinanceAccountDto
    {
        public int account_id { get; set; }
        public string account_name { get; set; }
        public string account_type { get; set; }
        public decimal initial_balance { get; set; }
        public decimal current_balance { get; set; }
        public string currency { get; set; }
    }
}

File: sources/nest-backend/src/account/dto/update-finance-account.dto.ts
import { IsNotEmpty, IsNumber, IsString, IsOptional } from 'class-validator';

export class UpdateFinanceAccountDto {
  @IsNotEmpty()
  @IsNumber()
  account_id: number;

  @IsOptional()
  @IsString()
  account_name?: string;

  @IsOptional()
  @IsString()
  account_type?: string;

  @IsOptional()
  @IsNumber()
  initial_balance?: number;

  @IsOptional()
  @IsNumber()
  current_balance?: number;

  @IsOptional()
  @IsString()
  currency?: string;
}
